systemFlake.nix

```diff
@@ -100,8 +102,10 @@ let
               })
 
               (if options ? nixpkgs then {
-                nixpkgs = {
-                  inherit (selectedNixpkgs) overlays config system;
+                nixpkgs.pkgs = import patchedChannel {
+                  inherit (host) system;
+                  overlays = sharedOverlays ++ (if (channelValue ? overlaysBuilder) then (channelValue.overlaysBuilder self.pkgs) else [ ]);
+                  config = channelsConfig // (channelValue.config or { }) // config.nixpkgs.config;
```

Comment:
@Pacman99 The condensed reason for this is `config.nixpkgs.config` to work as the nix-manual states.

However it odes import nixpkgs on every host and does not "re-use" the import of `self.pkgs`. Conceptually this is fine, since anything host-specific would have no natuarl pace in `self.pkgs` (which is system / channel specific).

However, I'm a bit vary that we would import nixpkgs for every single host (again). That doesn't scale very well if we have 5000 otherwise uniform hosts to update, for example.

I'd be very much in favor of this addition if it was possible to only trigger it (and opt to not re-use imports from `self.pkgs`) in case `config.nixpkgs.config` was meaningfully set.

I ping you since, maybe you might have an idea here how to accomplish this.

Reply:
Note that the `self.pkgs` import never actually happens, since Nix is lazy, and `self.pkgs` isn't used anywhere for a nixos configuration.

Like I said, detecting use of nixpkgs.config is not easy. The full configuration for a host would have to be evaluated separately to avoid infinite recursion (we can't check whether something is set in what we are currently evaluating if we might change the outcome of that check based on the result of that check).